src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -234,56 +239,61 @@ public int RemoveAdornments()
+                threadingContext.ThrowIfNotOnUIThread();
+
+                var languageServiceBroker = _languageService.Package.ComponentModel.GetService<ILanguageServiceBroker2>();
                 var asyncListenerProvider = _languageService.Package.ComponentModel.GetService<IAsynchronousOperationListenerProvider>();
                 var asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.DocumentOutline);
                 var editorAdaptersFactoryService = _languageService.Package.ComponentModel.GetService<IVsEditorAdaptersFactoryService>();
 
-                threadingContext.ThrowIfNotOnUIThread();
-
                 // Assert that the previous Document Outline Control and host have been freed. 
-                Contract.ThrowIfFalse(_documentOutlineControl is null);
+                Contract.ThrowIfFalse(_documentOutlineView is null);
                 Contract.ThrowIfFalse(_documentOutlineViewHost is null);
 
-                _documentOutlineControl = new DocumentOutlineControl(
-                    languageServiceBroker, threadingContext, asyncListener, editorAdaptersFactoryService, _codeWindow);
+                var viewTracker = new VsCodeWindowViewTracker(_codeWindow, threadingContext, editorAdaptersFactoryService);
+                var textView = viewTracker.GetActiveView();
+                _documentOutlineView = new DocumentOutlineView(
+                    threadingContext, viewTracker,
+                    new DocumentOutlineViewModel(textView, threadingContext, languageServiceBroker, asyncListener));
 
                 _documentOutlineViewHost = new ElementHost
                 {
                     Dock = DockStyle.Fill,
-                    Child = _documentOutlineControl
+                    Child = _documentOutlineView
                 };
 
                 phwnd = _documentOutlineViewHost.Handle;
-                ppCmdTarget = null;
 
                 Logger.Log(FunctionId.DocumentOutline_WindowOpen, logLevel: LogLevel.Information);
-
-                return VSConstants.S_OK;
             }
 
             int IVsDocOutlineProvider.ReleaseOutline(IntPtr hwnd, IOleCommandTarget pCmdTarget)
             {
                 var threadingContext = _languageService.Package.ComponentModel.GetService<IThreadingContext>();
                 threadingContext.ThrowIfNotOnUIThread();
 
-                if (_documentOutlineControl is not null &&
+                if (_documentOutlineView is not null &&
                     _documentOutlineViewHost is not null)
                 {
                     _documentOutlineViewHost.SuspendLayout();
-                    _documentOutlineControl.Dispose();
-                    _documentOutlineControl = null;
+                    _documentOutlineView.Dispose();
+                    _documentOutlineView = null;
                     _documentOutlineViewHost.Child = null;
                     _documentOutlineViewHost.Parent = null;
```

Comment:
Is nulling these out really necessary given we're disposing and nulling out the field immediately afterwards?

Reply:
i'm not sure here.  This was the code from before, so i kept it.  I'm hoping wpf is not holding onto things, but i'm not savvy enough to really know for certain.  @jmarolf @ryzngard ?